engine/servermaster/executor_manager.go

```diff
@@ -169,24 +169,39 @@ func (e *ExecutorManagerImpl) RegisterExec(info *model.NodeInfo) {
 // AllocateNewExec allocates new executor info to a give RegisterExecutorRequest
 // and then registers the executor.
 func (e *ExecutorManagerImpl) AllocateNewExec(req *pb.RegisterExecutorRequest) (*model.NodeInfo, error) {
-	log.Info("allocate new executor", zap.Stringer("req", req))
+	executor := req.Executor
+	log.Info("allocate new executor", zap.Stringer("executor", executor))
 
 	e.mu.Lock()
-	info := &model.NodeInfo{
-		ID:         model.ExecutorID(e.idAllocator.NewString()),
-		Addr:       req.Address,
-		Capability: int(req.Capability),
+	var executorID model.ExecutorID
+	for {
+		executorID = generateExecutorID(executor.GetName())
+		if _, ok := e.executors[executorID]; !ok {
+			break
+		}
 	}
-	if _, ok := e.executors[info.ID]; ok {
-		e.mu.Unlock()
-		return nil, errors.ErrExecutorDupRegister.GenWithStackByArgs()
+	info := &model.NodeInfo{
+		ID:         executorID,
+		Addr:       executor.GetAddress(),
+		Name:       executor.GetName(),
+		Capability: int(executor.GetCapability()),
 	}
 	e.mu.Unlock()
 
 	e.RegisterExec(info)
 	return info, nil
 }
 
+func generateExecutorID(name string) model.ExecutorID {
+	var buf [4]byte
+	_, err := io.ReadFull(rand.Reader, buf[:])
+	if err != nil {
```

Comment:
I think it's better to use another random algorithm if `ReadFull` fail, instead of panic the `servermaster`.

Reply:
Changed to `math/rand`. https://github.com/pingcap/tiflow/pull/6864/commits/02ad0cfc6cea049c8cf00cf409c2bf71bbdc3f00